src/lib.rs

```diff
@@ -239,14 +258,26 @@ fn setup_bitcoind(
         log::info!("Creating a new watchonly wallet on bitcoind.");
         bitcoind.create_watchonly_wallet(&config.main_descriptor)?;
         log::info!("Watchonly wallet created.");
-    } else {
+    } else if !is_remote {
         #[cfg(windows)]
         if !cfg!(test) && !wo_path.exists() {
             return Err(StartupError::NoWatchonlyInDatadir);
         }
     }
     log::info!("Loading our watchonly wallet on bitcoind.");
-    bitcoind.maybe_load_watchonly_wallet()?;
+    if is_remote {
+        if let Err(e) = bitcoind.maybe_load_watchonly_wallet() {
+            if e.is_wallet_not_found() {
+                log::info!("Loading legacy wallet.");
+                bitcoind = BitcoinD::new(bitcoind_config, wo_path_str)?;
+                bitcoind.maybe_load_watchonly_wallet()?;
+            } else {
+                Err(e)?
+            }
+        }
+    } else {
+        bitcoind.maybe_load_watchonly_wallet()?;
```

Comment:
nit: I think you could have a single call to `maybe_load_watchonly_wallet()` after the `if {} else {}` block.

Reply:
+1. I think you should `match` on the call to `maybe_load_watchonly_wallet` and only check `is_remote` in case of failure.